erpnext_shipping/erpnext_shipping/utils.py

```diff
@@ -46,6 +48,37 @@ def validate_address(address):
 		frappe.throw(_("Please add a valid pincode in Address {0}.").format(address.address_title))
 
 
+def validate_parcels(doc, method=None):
+	parcels = doc.get("shipment_parcel")
+
+	if parcels:
+		for parcel in parcels:
+			if (parcel.get("length") or 0) < 1:
+				frappe.throw(_("Parcel length must be greater than 1."))
+			if (parcel.get("width") or 0) < 1:
+				frappe.throw(_("Parcel width must be greater than 1."))
+			if (parcel.get("height") or 0) < 1:
+				frappe.throw(_("Parcel height must be greater than 1."))
```

Comment:
```suggestion
	if doc.docstatus != 0:
		return

	for parcel in doc.shipment_parcel:
		if (parcel.get("length") or 0) < 1:
			frappe.throw(_("Parcel length must be greater than 1."))
		if (parcel.get("width") or 0) < 1:
			frappe.throw(_("Parcel width must be greater than 1."))
		if (parcel.get("height") or 0) < 1:
			frappe.throw(_("Parcel height must be greater than 1."))
```

- I think we can rely on `doc.shipment_parcel` containing at least an empty list.
- By checking the docstatus, avoid the scenario where we try to update tracking information on a submitted **Shipment** but get an error because the dimensions are wrong and cannot be changed anymore.